src/docfx/lib/JsonUtility.cs

```diff
@@ -345,9 +347,24 @@ private static (Range, string message, string path) ParseException(Exception ex)
             if (match.Success)
             {
                 var range = new Range(int.Parse(match.Groups[3].Value), int.Parse(match.Groups[4].Value));
-                return (range, match.Groups[1].Value, match.Groups[2].Value);
+                return (range, RewriteErrorMessage(match.Groups[1].Value), match.Groups[2].Value);
+            }
+
+            match = Regex.Match(ex.Message, "^([\\s\\S]*)\\sPath '(.*)'.$");
+            if (match.Success)
+            {
+                return (default, RewriteErrorMessage(match.Groups[1].Value), match.Groups[2].Value);
+            }
+            return (default, RewriteErrorMessage(ex.Message), null);
+        }
+
+        private static string RewriteErrorMessage(string message)
+        {
+            if (message.StartsWith("Error reading string. Unexpected token"))
+            {
+                return "Expected type String, please input String or type compatible with String.";
```

Comment:
only works for expecting string? what about other expecting case or add a other type mismatch test case?

Reply:
https://github.com/dotnet/docfx/blob/8569891a307da86ae7704810d59b475a6fcff4cc/test/docfx.Test/lib/JsonUtilityTest.cs#L331
Covered here